examples/wave-eager-mpi.py

```diff
@@ -139,7 +139,17 @@ def main(snapshot_pattern="wave-eager-{step:04d}-{rank:04d}.pkl", restart_step=N
         t = restart_data["t"]
         istep = restart_step
         assert istep == restart_step
-        fields = restart_data["fields"]
+        restart_fields = restart_data["fields"]
+        old_order = restart_data["order"]
+        if old_order != order:
+            old_discr = EagerDGDiscretization(actx, local_mesh, order=old_order,
+                                              mpi_communicator=comm)
```

Comment:
Not loving that this still uses the old-timey `EagerDGDiscretization`, but OK with me.